client/lib/purchases/assembler.js

```diff
@@ -65,14 +60,12 @@ function createPurchaseObject( purchase ) {
 		refundText: purchase.refund_text,
 		refundPeriodInDays: purchase.refund_period_in_days,
 		renewDate: purchase.renew_date,
-		// only generate a moment if `renewDate` is present and positive
-		renewMoment:
-			purchase.renew_date && purchase.renew_date > '0' ? i18n.moment( purchase.renew_date ) : null,
+		// only consider valid if `renewDate` is present and positive
+		renewDateValid: Boolean( purchase.renew_date ) && purchase.renew_date > '0',
```

Comment:
This feels suboptimal to me - having another field just to indicate whether a certain field is valid. I wonder if it makes sense to erase the `renewDate` field if it's not valid - the purchase assembler seems like a good place to do that.

Reply:
Agreed. Now that the date format has been normalised (thank you to @jsnajdr for implementing and to you for reviewing!), we no longer need to do this 👍